### DISCUSSION TITLE
Problems with asynchronous queries

### DISCUSSION THREAD

**Discussion by Lakremon** [mikro-orm/mikro-orm#4848] [General]
**Describe the bug**
My application is built on asynchronous architecture. There are different parts of the application that work independently of each other. The problem occurs when 2 parts of the application begin to make a request to the database at the same time.

To me it looks like MicroORM is designed to work with asynchronous requests. By enabling query logging, I can see how the ORM combines queries into transactions. And queries compiled in this way are executed on the database side. The problem arises in parsing query results, because queries do not continue executing the program but simply wait indefinitely for the result. All I see in the ORM logs are endless attempts to execute the same pool of requests.

**Stack trace**
```
[query] begin
[query] delete from `ltg_statistics` where `id` in ('676', '1236', '2009', '3033', '3895', '7132', '7987', '8230', '8663', '8826', '10217', '677', '1237', '2010', '3034', '3896', '7133', '7988', '8664', '8827', '10218', '678', '1238', '2011', '3035', '3897', '7134', '7989', '8665', '8828', '10219', '10855', '10889') [took 201 ms]
[query] delete from `ltg_events` where `id` in ('237', '238', '239') [took 178 ms]
[query] select `s0`.*, `l1`.`id` as `ltg_ticket_id` from `sb_inventories` as `s0` left join `ltg_tickets` as `l1` on `s0`.`id` = `l1`.`sb_inventory_id` where `s0`.`id` = '4209' limit 1 [took 585 ms, 1 result]
[query] delete from `ltg_categories` where `id` in ('158') [took 178 ms]
[query] commit
[query] begin
[query] select `s0`.*, `l1`.`id` as `ltg_ticket_id` from `sb_inventories` as `s0` left join `ltg_tickets` as `l1` on `s0`.`id` = `l1`.`sb_inventory_id` where `s0`.`id` = '525406489' limit 1 [took 240 ms, 1 result]
[query] delete from `ltg_statistics` where `id` in ('676', '1236', '2009', '3033', '3895', '7132', '7987', '8230', '8663', '8826', '10217', '677', '1237', '2010', '3034', '3896', '7133', '7988', '8664', '8827', '10218', '678', '1238', '2011', '3035', '3897', '7134', '7989', '8665', '8828', '10219', '10855', '10889') [took 200 ms]
[query] delete from `ltg_events` where `id` in ('237', '238', '239') [took 201 ms]
[query] delete from `ltg_categories` where `id` in ('158') [took 177 ms]
[query] commit
```

**To Reproduce**
1. Create an asynchronous JS application such that it executes several async functions
2. Make an endless loop to creqte queue
```
setInterval(async () => {
        ...
        const ltgTickets = await em.find(LtgTickets, {
            ltgEvent: {
                startDateTime: {$gt: new Date()},
            },
            id: {$nin: toEdit.concat(inEditProcess)},
        });
        ...
}, 2 * 1000);
```
3. Use in other part of application like this:
```
...
await em.removeAndFlush(toDelete);
...
```

**Expected behavior**
Both requests will be executed and the program will continue to run.

**Additional context**

**Versions**

| Dependency | Version |
| node | 18.17.1 |
| typescript | 4.4.4 |
| mikro-orm | 5.7.13 |

**Comment by B4nan** [comment]
> | mikro-orm | 5.7.13 |

This is pretty old, use the latest version to be sure if it's not something already fixed.

> Create an asynchronous JS application such that it executes several async functions

This is way too sparse. You will need to provide complete but minimal reproduction if you want help with this. Generally speaking, it is not something the ORM was designed for, your memory will grow and probably overflow at some point if you use one global context. You can't just load things to the context forever and expect them to get freed somehow - they won't.

**Comment by Lakremon** [comment]
Thank you, sir! It works fine with version 5.8.10!

However I do not agree "it is not something the ORM was designed for". If you implement this feature - it should work! Are you ok if I add a test-case for this? Where is the best place to add? /mikro-orm/tests/features/async-queries is ok?

**Reply by B4nan** [reply]
Implement what feature?
